### ISSUE TITLE
Open intervals for minrun-maxrun in database

### ISSUE THREAD

**Issue by gonzaponte** [next-exp/IC#364]
When we search for data in the database based on some run number, we look for those rows between `MinRun` and `MaxRun` with an open interval. For example:

```
select * from ChannelMask where MinRun < {3} and {3} < MaxRun
```

Since the data runs have never matched those numbers so far, there has never been a problem with that. Do we want to change it to be a right-open interval like `[MinRun, MaxRun)`? I've been struggling to figure out why a test failed because of this subtlety.

**Comment by jacg** [comment]
Sticking to the strong Python convention of semi-open intervals would certainly conform to the principle of least surprise.

What's really shocking in this case is that it's an *open* interval: usually deviations from the semi-open convention tend towards closed intervals, so open seems very surprising indeed.

The fact that you lost time on undertanding a test failure because of it, is an indication that it should be changed.

**Comment by jmbenlloch** [comment]
I agree there is an error there, but I think the interval has to be closed `[MinRun, MaxRun]` due to two things:

1- That is the way data is stored in the DB now. For instance:

```
mysql> select distinct(MinRun), MaxRun from ChannelMapping;
+--------+--------+
| MinRun | MaxRun |
+--------+--------+
|      0 |   1273 |
|   1274 |   1754 |
|   1755 |   1765 |
|   1766 |   1978 |
|   1979 | 100000 |
|   1979 |   3292 |
|      0 | 100000 |
|   3293 |   3913 |
|   3914 | 100000 |
+--------+--------+
9 rows in set (0,91 sec)
```

2- We arre using the same DB as the ART software, if we change the DB to use semiopen intevals, we may break compatibility with ART.

So, in my opinion, we sould put: `select * from ChannelMask where MinRun <= {3} and {3} <= MaxRun`.

**Comment by gonzaponte** [comment]
Ok. You mean that `MaxRun` is always the next `MinRun` minus one. Closed intervals make sense here.

**Comment by gonzaponte** [comment]
This should't have been closed. There is no PR fixing it.

**Comment by jjgomezcadenas** [comment]
Sorry, I don't get it. I think it is agreed that we cannot use semi-open interval in the case of the DB

**Comment by gonzaponte** [comment]
We have agreed on the solution, but it is still not implemented.

**Comment by jacg** [comment]
This was supposed to have been closed automatically by the merge of #450. This did not happen because the magical incantation was written in a PR comment. GitHub only looks for them in the PR description (the top comment) and in commit messages.
